Sklep/MainWindow.cs

```diff
@@ -35,17 +48,75 @@ private void BarcodeScanner_CodeScanned(object sender, string code)
             {
                 barCodeTextBox.Text = code;
             }));
+            cashRegisterBeep.Play();
+            addProductToList(code);
+
         }
+        private void addProductToList(string scannedBarcode)
+        {
+            if (scannedBarcode == "") return;
+
+            using (var context = new DatabaseContext())
```

Comment:
Tutaj można dość mocno zoptymalizować kod, dla wielu takich samych produktów. Spójrz, że jeśli produkt jest już w koszyku, to kolejne zeskanowanie wykona zapytanie do bazy danych, a potem nie użyje jego wyniku `(position != null)`

Lepiej byłoby najpierw sprawdzić, czy taki produkt jest już w koszyku, jeśli tak to zwiększyć ilość i fajrancik, i dopiero w przeciwnym wypadku wysłać zapytanie do bazy o ten produkt.